packages/twenty-front/src/modules/prefetch/hooks/usePrefetchedData.ts

```diff
@@ -0,0 +1,43 @@
+import { useRecoilValue } from 'recoil';
+
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { ALL_FAVORITES_QUERY_KEY } from '@/prefetch/query-keys/AllFavoritesQueryKey';
+import { ALL_VIEWS_QUERY_KEY } from '@/prefetch/query-keys/AllViewsQueryKey';
+import { prefetchIsLoadedFamilyState } from '@/prefetch/states/prefetchIsLoadedFamilyState';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKeys';
+
+export const usePrefetchedData = <T extends ObjectRecord>(
+  prefetchKey: PrefetchKey,
+) => {
+  const isDataPrefetched = useRecoilValue(
+    prefetchIsLoadedFamilyState(prefetchKey),
+  );
+  const prefetchQueryKey = computeQueryKey(prefetchKey);
+
+  const { records } = useFindManyRecords<T>({
+    skip: !isDataPrefetched,
+    ...prefetchQueryKey,
+    useRecordsWithoutConnection: true,
+  });
+
+  return {
+    isDataPrefetched,
+    records,
+  };
+};
+
+const computeQueryKey = (prefetchKey: PrefetchKey) => {
```

Comment:
Could we store this in DB ? Maybe we could rely on a hardcoded JSON config for now ? The function would just need to be passed a JSON config that would abstract away the problem.